delegator/src/sign/sign.service.ts

```diff
@@ -97,26 +93,24 @@ export class SignService {
     return tx.getRawTransaction()
   }
 
-  validateTransaction(tx) {
-    // FIXME remove simple whiteListing settings and setup db
-    const contractList = {
-      '0x93120927379723583c7a0dd2236fcb255e96949f': {
-        methods: ['increment()', 'decrement()'],
-        reporters: ['0x260836ac4f046b6887bbe16b322e7f1e5f9a0452']
+  async validateTransaction(tx) {
+    const result = await this.prisma.contract.findMany({
+      where: {
+        address: tx.to,
+        Reporter: {
+          some: {
+            address: tx.from
+          }
+        },
+        Function: {
+          some: {
+            encodedName: tx.input.substring(0, 11)
+          }
+        }
       }
-    }
-
-    if (!(tx.to in contractList)) {
-      throw new DelegatorError(DelegatorErrorCode.InvalidContract)
-    }
-    if (!contractList[tx.to].reporters.includes(tx.from)) {
-      throw new DelegatorError(DelegatorErrorCode.InvalidReporter)
-    }
-    const isAllowedMethod = contractList[tx.to].methods.some(
-      (method) => this.encryptMethodName(method) == tx.input
-    )
-    if (!isAllowedMethod) {
-      throw new DelegatorError(DelegatorErrorCode.InvalidMethod)
+    })
+    if (result.length == 0) {
+      throw new DelegatorError(DelegatorErrorCode.InvalidTransaction)
```

Comment:
I would recommend to change the name of the error. The exception is thrown when there is not a transaction corresponding to the search criteria, therefore the issue is not caused by `invalid transaction`.

How about changing the error name to `NoTransactionFound` instead?

Reply:
I think the error name `NoTransactionFound` might be misleading. 

This exception is thrown when the input `transaction` is not approved by the current whitelist configuration. The function checks it by looking for `contract` by filtering it with `reporter address` and `functionName`.  

How about changing it to `NotApprovedTransaction`?

Reply:
Aha, you are right! I like your suggestion! Let's go with that :)